gradoop-data-integration/src/main/java/org/gradoop/dataintegration/importer/csv/MinimalCSVImporter.java

```diff
@@ -0,0 +1,169 @@
+   * will set as the property names for each column.
+   * @param path the path to the csv file
+   * @param tokenSeperator the token delimiter of the csv file
+   * @param config GradoopFlinkConfig
+   * @param charset the charset used in the csv file
+   */
+  public MinimalCSVImporter(String path, String tokenSeperator, GradoopFlinkConfig config,
+          String charset) {
+    this.path = path;
+    this.tokenSeparator = tokenSeperator;
+    this.config = config;
+    this.charset = charset;
+  }
+
+  /**
+   * Import each row of the file as a vertex. If no column property names are set,
+   * read the first line of the file as header and set this values as column names.
+   * @param checkReoccurringHeader if each row of the file should be checked for reocurring of
+   * the column property names.
+   * @return the imported vertices
+ * @throws IOException if an error occurred while open the stream
+   */
+  public DataSet<ImportVertex<Long>> importVertices(boolean checkReoccurringHeader)
+          throws IOException {
+    if (columnNames == null) {
+      return readCSVFile(readHeaderRow(), checkReoccurringHeader);
+    } else {
+      return readCSVFile(columnNames, checkReoccurringHeader);
+    }
+  }
+
+  /**
+   * Read the vertices from a csv file.
+   *
+   * @param propertyNames list of the property identifier name
+   * @param checkReoccurringHeader if each row of the file should be checked for reocurring of
+   * the column property names.
+   * @return DateSet of all vertices from one specific file.
+   */
+  public DataSet<ImportVertex<Long>> readCSVFile(List<String> propertyNames,
+          boolean checkReoccurringHeader) {
+
+    DataSet<Properties> lines = config.getExecutionEnvironment()
+    .readTextFile(path)
+    .map(new RowToVertexMapper(path, tokenSeparator, propertyNames, checkReoccurringHeader))
+    .filter(new FilterNullValuesTuple<>());
+
+    return DataSetUtils.zipWithUniqueId(lines).map(new CreateImportVertexCSV<>());
+  }
+
+  /**
+   * Read the fist row of a csv file and put each the entry in each column in a list.
+   * @return the property names
+ * @throws IOException if an error occurred while open the stream
+   */
+  public List<String> readHeaderRow() throws IOException {
+    try (final BufferedReader reader =
+              new BufferedReader(new InputStreamReader(new FileInputStream(path),
+                      charset))) {
+      String headerLine = reader.readLine();
+      if (headerLine == null) {
+        LOG.error("The file do not contain any rows.");
+        throw new NullPointerException();
+      }
+      headerLine = headerLine.substring(0, headerLine.length());
+      String[] headerArray;
+      headerArray = headerLine.split(tokenSeparator);
+
+      return Arrays.asList(headerArray);
+    } catch (IOException ex) {
+      LOG.error("I/O Error occurred while trying to open a stream to: '" +
+        path + "'.");
+      throw new IOException();
```

Comment:
Change the exception to the following: 
```
throw new IOException("I/O Error occurred while trying to open a stream to: '" +
        path + "'." + ex);
```

Reply:
Better put `ex` as second parameter of constructor as a cause. 
```
throw new IOException("I/O Error occurred while trying to open a stream to: '" + path + "'.", ex);
```